dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -38,6 +37,116 @@ TypeInfo TypeFromNumpyStr(const std::string &format) {
   DALI_FAIL("Unknown Numpy type string");
 }
 
+inline void SkipSpace(const char*& ptr) {
+  while (::isspace(*ptr))
+    ptr++;
+}
+
+template <size_t N>
+void Skip(const char*& ptr, const char (&what)[N]) {
+  DALI_ENFORCE(!strncmp(ptr, what, N - 1),
+               make_string("Expected \"", what, "\" but got \"", ptr, "\""));
+  ptr += N - 1;
+}
+
+template <size_t N>
+bool TrySkip(const char*& ptr, const char (&what)[N]) {
+  if (!strncmp(ptr, what, N-1)) {
+    ptr += N - 1;
+    return true;
+  } else {
+    return false;
+  }
+}
+
+
+template <size_t N>
+void SkipFieldName(const char*& ptr, const char (&name)[N]) {
+  SkipSpace(ptr);
+  Skip(ptr, "'");
+  Skip(ptr, name);
+  Skip(ptr, "'");
+  SkipSpace(ptr);
+  Skip(ptr, ":");
+  SkipSpace(ptr);
+}
+
+std::string ParseStringValue(const char*& input, char delim_start = '\'', char delim_end = '\'') {
+  DALI_ENFORCE(*input++ == delim_start, make_string("Expected \'", delim_start, "\'"));
+  std::string out;
+  for (; *input != '\0'; input++) {
+    if (*input == '\\') {
+      switch (*++input) {
+        case '\'':
+          out += '\\';
+          break;
+        case '\t':
+          out += '\t';
+          break;
+        case '\n':
+          out += '\n';
+          break;
+        case '\"':
+          out += '\"';
+          break;
+        default:
+          out += '\\';
+          out += *input;
+          break;
+      }
+    } else if (*input == delim_end) {
+      break;
+    } else {
+      out += *input;
+    }
+  }
+  DALI_ENFORCE(*input++ == delim_end, make_string("Expected \'", delim_end, "\'"));
+  return out;
+}
+
+void ParseHeaderMetadata(NumpyParseTarget& target, const std::string &header) {
+  const char* hdr = header.c_str();
+  SkipSpace(hdr);
+  Skip(hdr, "{");
+  SkipFieldName(hdr, "descr");
+  auto typestr = ParseStringValue(hdr);
+    // < means LE, | means N/A, = means native. In all those cases, we can read
+  bool little_endian = (typestr[0] == '<' || typestr[0] == '|' || typestr[0] == '=');
+  DALI_ENFORCE(little_endian, "Big Endian files are not supported.");
+  target.type_info = TypeFromNumpyStr(typestr.substr(1));
+
+  SkipSpace(hdr);
+  Skip(hdr, ",");
+  SkipFieldName(hdr, "fortran_order");
+  if (TrySkip(hdr, "True")) {
+    target.fortran_order = true;
+  } else if (TrySkip(hdr, "False")) {
+    target.fortran_order = false;
+  } else {
+    DALI_FAIL("Cannot read an array stored in Fortran order.");
```

Comment:
I know this was changed, but isn't it:
```suggestion
    DALI_FAIL("Cannot parse the Fortran order [of an array].");
```

Reply:
yeah, it is. My bad. I applied the fix without really reading it.